app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/CameraUtils.kt

```diff
@@ -0,0 +1,209 @@
+package com.example.android.architecture.blueprints.todoapp.data
+
+import android.content.Context
+import android.content.res.Resources
+import android.graphics.*
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.Drawable
+import com.example.android.architecture.blueprints.todoapp.R
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.FileOutputStream
+import java.io.InputStream
+import java.util.*
+
+fun getCircleBitmap(color: Int, size: Int, activityContext: Context): Drawable {
+    val a = Bitmap.createBitmap(size, size, Bitmap.Config.ARGB_8888)
+    val canvas = Canvas(a)
+    val paint = Paint()
+    paint.isAntiAlias = true
+    paint.color = color
+    paint.style = Paint.Style.FILL
+    canvas.drawBitmap(a, 0f, 0f, paint)
+    canvas.drawCircle(size / 2f, size / 2f, size / 2f, paint)
+    return BitmapDrawable(activityContext.resources, a)
+}
+
+ fun saveStreamToFile(inputStream: InputStream,
+                             resources: Resources,
+                             activity: Context): Pair<String?, String?> {
+    var outputPath: String? = null
+    var originalOutputPath: String? = null
+    try {
+        val filesDir = activity.getExternalFilesDir(null)
+        val photo: Bitmap?
+        val options = BitmapFactory.Options()
+        options.inMutable = true
+        options.inDensity = 5000
+        options.inScreenDensity = 5000
+        options.inTargetDensity = 5000
+
+        photo = BitmapFactory.decodeStream(inputStream, null, options)
+
+        inputStream.close()
+
+        val originImageFile = File(filesDir, System.currentTimeMillis().toString() + "_.jpg")
+        if (originImageFile.createNewFile()) {
+            val fos = FileOutputStream(originImageFile)
+            photo!!.compress(Bitmap.CompressFormat.JPEG, 100, fos)
```

Comment:
photo?.compress(Bitmap.CompressFormat.JPEG, 100, fos)

better to keep null-safety